### PR TITLE
fix: braze dedup for non-billable attributes

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

Fixes for dedup utility for allowing non-billable attributes, previously if only non-billable attributes were only available in traits those were filtered. Now these are allowed

## What is the related Linear task?

Resolves INT-2071

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by utsabc** [comment]
@coderabbitai review

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3320/test-report.html

File: src/v0/destinations/braze/util.js [hunk 1]
```diff
@@ -284,12 +284,17 @@ const BrazeDedupUtility = {
...
+          deduplicatedUserData[key] = userData[key];
+        }
+      });
     }
 
-    keys.forEach((key) => {
-      if (!_.isEqual(userData[key], storedUserData[key])) {
+    // add non billable attributes back to the deduplicated user object
+    BRAZE_NON_BILLABLE_ATTRIBUTES.forEach((key) => {
+      if (isDefinedAndNotNull(userData[key])) {
```
Comment (sanpj2292) [inline src/v0/destinations/braze/util.js:297]: Should also forward non-billable attributes that are null as braze accepts null attributes ?

Comment (utsabc) [reply src/v0/destinations/braze/util.js]: Addressed

**Review by sanpj2292 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**